tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala

```diff
@@ -0,0 +1,148 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.tasks.vc
+
+import java.nio.file.Path
+
+import dagr.commons.CommonsDef.yieldAndThen
+import dagr.commons.io.Io
+import dagr.core.config.Configuration
+import dagr.core.execsystem.{Cores, Memory, ResourceSet}
+import dagr.core.tasksystem._
+import dagr.tasks.DagrDef.{FilePath, PathToBam, PathToFasta, PathToVcf}
+import dagr.tasks.DataTypes.Vcf
+import htsjdk.samtools.SamReaderFactory
+
+import scala.collection.mutable.ListBuffer
+
+/**
+  * The VarDictJava directory structure is assumed to be as follows:
+  *   - Root directory: /path/to/git/VarDictJava
+  *   - Perl scripts:   /path/to/git/VarDictJava/VarDict
+  *   - Startup script: /path/to/git/VarDictJava/build/install/VarDict/bin/VarDict
+  */
+object VarDictJava extends Configuration {
+  /** Config key for the VarDictJava directory.  This should be the root project directory. */
+  val RootDirConfigKey = "vardictjava.dir"
+
+  /** Path to the VarDictJava directory. */
+  lazy val RootDir = configure[Path](RootDirConfigKey)
+
+  /** Path to the VarDictJava install directory with the VarDictJava startup script. */
+  lazy val VarDictJava = RootDir.resolve("build/install/VarDict/bin/VarDict")
+
+  /** Path to the VarDictJava bin directory with a multitude of perl scripts. */
+  lazy val BinDir = RootDir.resolve("VarDict")
+
+  /** Pulls a sample name out of a BAM file. */
+  private[vc] def extractSampleName(bam: PathToBam) : String = {
+    val in = SamReaderFactory.make().open(bam)
+    yieldAndThen(in.getFileHeader.getReadGroups.iterator().next().getSample) { in.close() }
+  }
+}
+
+/** Runs VarDictJava to produce its intermediate tabular format. */
+private class VarDictJava(tumorBam: PathToBam,
+                          bed: FilePath,
+                          ref: PathToFasta,
+                          tumorName: String,
+                          minimumAf: Double = 0.01,
+                          includeNonPf: Boolean = false,
+                          minimumQuality: Int = 0,
+                          maximumMismatches: Int = 8,
+                          minThreads: Int = 1,
+                          maxThreads: Int = 32,
+                          memory: Memory = Memory("8G")
+                         ) extends ProcessTask with VariableResources with PipeOut[Vcf] {
+  name = "VarDictJava"
+
+  override def pickResources(resources: ResourceSet): Option[ResourceSet] = {
+    resources.subset(minCores = Cores(minThreads), maxCores = Cores(maxThreads), memory = memory)
+  }
+
+  override def args: Seq[Any] = {
+    val vdScript = VarDictJava.VarDictJava
+    val buffer   = new ListBuffer[Any]()
+
+    buffer.appendAll(List(vdScript, "-G", ref, "-N", tumorName, "-b", tumorBam))
+    buffer.appendAll(List("-z", "-c", "1", "-S", "2", "-E", "3", "-g", "4"))
```

Comment:
Isn't `-z` supposed to take either `0` or `1` as an option value?

Also, can I suggest that you break these out onto multiple lines and add a short comment on each line describing what the option does?  Will save going back and forth to the usage all the time to remember what they all do.